src/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -533,6 +545,10 @@ void CodeGen::genHWIntrinsicJumpTableFallback(NamedIntrinsic            intrinsi
 
     emit->emitDataGenEnd();
 
+    // Ensure we aren't overwriting offsReg, baseReg, or nonConstImmReg
+    assert(offsReg != baseReg);
```

Comment:
Is there any guarantee that two temporary registers won't be the same? Do I need to set `isInternalRegDelayFree` to guarantee that?

Reply:
Temporary registers have identical live ranges so they cannot be the same. If their live ranges would be disjoint then you'd probably need only one temporary register to begin with.